### ISSUE TITLE
Fix flannel integration broken by #6921

### ISSUE THREAD

**Issue by jrajahalme** [cilium/cilium#7523]
#6921 seems to have broken flannel integration. Figure out why.

**Comment by aanm** [comment]
```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
4eb97b91843981b999f39e4acebf59681571f91f
566330e874424561dc4089865e299e5520c88d37
d65adf83f8be1d6d07f63a9b1e8a8197bfe7ae0e
```

**Comment by jrajahalme** [comment]
@aanm This is probably caused due to the way I removed the bpf redirects in the TPROXY PR. I guess that with flannel (and possibly with other integrations) we don't use the cilium_host/net veth pair for all traffic, which the new TPROXY iptables rules now somehow depend on. I'll fix this next week by reintroducing the bpf_redirects for the proxy traffic, maybe conditionally.

**Comment by joestringer** [comment]
I believe that the integration was fixed it v1.6, however the CI testing continues to be broken.

**Comment by joestringer** [comment]
This has been open for a long time and would usually be marked as stale, but it's assigned to someone so that mechanism did not work. I don't see a reason to keep it open.
